### ISSUE TITLE
Nested types cause "Unknown module or contract"

### ISSUE THREAD

**Issue by asbjornenge** [OCamlPro/liquidity#202]
Hi there!

I'm setting out to build a tezos smart contract backed app. I'm using liquidity and techelson via my own testrunner [sliq](https://github.com/asbjornenge/sliq) (just a cli passing contracts and tests to a docker image for exec).

I have an issues where when I try to use nested types I get `<unspecified>: Error: Unknown module or contract Appstore` from liquidity 🤔

Here is my `Appstore.reliq`:

```
type app = {
  name: string,
  owner: address,
};

type storage =  map(string, app);

let%entry main = (name: string, store) => {
  let amount = Current.amount();

  if (amount < 5.00tz) {
    Current.failwith("Not enough money, at least 5tz to vote");
  };

  ([], store);
};
```

If I modify `type storage =  map(string, app);`  with `type storage =  map(string, string);` it works as expected and tests are run.

I first reported this [issue](https://github.com/OCamlPro/techelson/issues/11) on the techelson repo and got redirected here 👍

Command & outout:
```
liquidity --no-annot --no-simplify --no-peephole /techel.liq Appstore.reliq -o /tmp/sliq.techel tests/Tests.reliq
Module Techel
Contract Appstore
<unspecified>: Error: Unknown module or contract Appstore
```

In the tests I'm using `Appstore.storage` etc.

Any help would be greatly appreciated 😄👍

Thanks!

**Comment by mebsout** [comment]
Sure. Can you attach all the files to reproduce the issue ?

**Comment by asbjornenge** [comment]
I was unable to attache .reliq files, but have pasted them below 👍

Appstore.reliq
```
type app = {
  name: string,
  owner: address,
};

type storage =  map(string, app);

let%entry main = (name: string, store) => {
  let amount = Current.amount();

  if (amount < 5.00tz) {
    Current.failwith("Not enough money, at least 5tz to vote");
  };

  ([], store);
};
```

Tests.reliq
```
let nothing: (list(operation), unit) = ([], ()) /* Creates a storage for Multi with one administrator. */;

let admins = Set([tz1YLtLqD1fWHthSVHPD116oYvsd4PTAHUoa]);

let init_storage = (name: string) => 
  Map([("name", "test")]);

let deploy_account_op = (amount: tez): (operation, address) => {
  let delegate: option(key_hash) = None;
  Account.create(
    ~manager=tz1YLtLqD1fWHthSVHPD116oYvsd4PTAHUoc,
    ~delegate,
    ~delegatable=true,
    ~amount,
  );
}; 

let deploy_contract_op = (storage: Appstore.storage): (operation, address) => {
  let delegate: option(key_hash) = None;
  Contract.create(
    ~manager=tz1YLtLqD1fWHthSVHPD116oYvsd4PTAHUoc,
    ~delegate,
    ~delegatable=true,
    ~spendable=false,
    ~amount=0tz,
    ~storage,
    ~code=(contract Appstore),
  );
};

type storage = unit;

let%entry test = (_param: unit, _storage: unit) => {
  let (fund_op, address) = deploy_account_op(10tz);
  let (main_op, main) = deploy_contract_op(init_storage("eple"));
  Techel.apply_operations([fund_op, main_op]);

  Techel.print_stack();

  let balance = Techel.get_balance(address);
  if (balance != 10tz) {
    failwith("balance should be 10tz");
  };

  let storage =
    switch (Techel.get_storage([%type: Appstore.storage], main)) {
    | Some(storage) => storage
    | None => failwith("can't retrieve contract's storage")
    };

  Techel.print_stack();

  nothing;
};
```

Techel.liq
```
external get_balance :
    [%stack: address] -> [%stack: tez]
    = "GET_BALANCE"
external get_storage :
    [%type: 'a] -> [%stack: address] -> [%stack: 'a option]
    = "GET_STORAGE"
external apply_operations :
    [%stack: operation list] -> unit
    = "APPLY_OPERATIONS"
external start_set_source :
    [%stack: address] -> unit
    = "SET_SOURCE { #"
external end_set_source :
    unit -> unit
    = "}"
external must_fail :
    [%stack: operation] -> [%stack: operation]
    = "{ NONE string ; MUST_FAIL string }"
external print_stack :
    unit -> unit
    = "PRINT_STACK"
external step :
    unit -> unit
    = "STEP"
```

**Comment by mebsout** [comment]
Thanks for the bug report. This will be fixed when #198 is merged.

**Comment by asbjornenge** [comment]
@mebsout excellent, thanks 👍
